src/Ship.js

```diff
@@ -0,0 +1,30 @@
+class Ship {
+  constructor(itinerary){
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+  };
+  setSail() {
+    const itinerary = this.itinerary;
+    const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+    if (currentPortIndex === (itinerary.ports.length -1)){
+      throw new Error('End of itinterary reached');
+    };
```

Comment:
The two variables on line 9 & 10 are only used once, so it makes more sense to remove them and reference the values directly where you were in the code:
```suggestion
    if (this.itinerary.ports.indexOf(this.currentPort) === (this.itinerary.ports.length -1)) {
      throw new Error('End of itinterary reached');
    };
```